src/viam/sdk/components/private/board_server.cpp

```diff
@@ -174,6 +177,51 @@ ::grpc::Status BoardServer::GetDigitalInterruptValue(
     return ::grpc::Status();
 }
 
+::grpc::Status BoardServer::StreamTicks(
+    ::grpc::ServerContext* context,
+    const ::viam::component::board::v1::StreamTicksRequest* request,
+    ::grpc::ServerWriter<::viam::component::board::v1::StreamTicksResponse>* writer) {
+    if (!request) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [Board::StreamTicks] without a request");
+    };
+
+    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
+    if (!rb) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    const std::shared_ptr<Board> board = std::dynamic_pointer_cast<Board>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    const std::shared_ptr<std::queue<Board::tick>> ticks;
+    const std::vector<std::string> digital_interrupt_names(request->pin_names().begin(),
+                                                           request->pin_names().end());
+
+    board->stream_ticks(digital_interrupt_names, ticks, extra);
+
+    ::viam::component::board::v1::StreamTicksResponse response;
+    while (true) {
+        if (!ticks->empty()) {
+            const Board::tick tick = ticks->front();
+            ticks->pop();
+            response.set_pin_name(tick.pin_name);
+            response.set_high(tick.high);
+            response.set_time(tick.time);
+            writer->Write(response);
```

Comment:
There is some extra copying goign on here. You are copying the element out of the front of the queue, then destroying the copy in the queue, then copying into the response. I think you can do fewer copies by keeping a reference to `front`, moving items out of it into the response, the popping at the end:
```
const auto& tick = ticks->front();
response.set_pin_name(std::move(tick.pin_name));
response.set_high(std::move(tick.high));
response.set_time(std::move(tick.time));
ticks->pop();
writer->Write(response);
```

Reply:
linter complained that we don't need the `std::move` but otherwise done.